web/client/selectors/controls.js

```diff
@@ -2,5 +2,21 @@ const {get} = require('lodash');
 
 module.exports = {
     queryPanelSelector: (state) => get(state, "controls.queryPanel.enabled"),
-    wfsDownloadAvailable: state => !!get(state, "controls.wfsdownload.available")
+    wfsDownloadAvailable: state => !!get(state, "controls.wfsdownload.available"),
+    cssStateSelector: state => {
+        const browser = get(state, "browser.mobile") && ' mobile' || ' desktop';
+
+        const drawer = get(state, "controls.drawer.enabled") && ' mapstore-drawer-open' || '';
+
+        const featuregrid = get(state, "featuregrid.open") && ' mapstore-featuregrid-open' || '';
+        const queryPanel = get(state, "controls.queryPanel.enabled") && ' mapstore-queryPanel-open' || '';
+
+        const annotation = get(state, "controls.annotations.enabled") && ' mapstore-annotation-open' || '';
+        const metadataexplorer = get(state, "controls.metadataexplorer.enabled") && ' mapstore-metadataexplorer-open' || '';
+        const identify = get(state, "mapInfo.requests") && state.mapInfo.requests.length > 0 && ' mapstore-identify-open' || '';
+
+        const rightPanel = metadataexplorer || annotation || identify ? ' mapstore-right-panel-open' : '';
```

Comment:
Get Feature Info panel is not well supported (because it's width is responsive). In addition any additional panel (even for custom plugins) should modify this selector. And any of these class is in css, and so resizable panel will not work.

IMHO we should use a higher level solution to this. Personally I don't like this solution because it is not pluggable at all, only makes the demo work. 
MS2 is framework to do custom applications, that typically will have some custom panels on the map. This solution do not allow the usage with custom panels.